<a href="https://colab.research.google.com/github/kportella/Challenge-Alura-Data-Science-Alura-Voz/blob/main/Challenge_Data_Science_Alura_Alura_Voz.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Challenge Data Science Alura - Alura Voz
## Author: Kauê Alã Portella
## Email: portellakaue@gmail.com
## Linkedin: www.linkedin.com/in/kaueportella
## Github: https://github.com/kportella

In [74]:
import pandas as pd
import numpy as np

In [75]:
pd.set_option("display.max_columns",100)

# Primeira Semana

## Tratamento de Dados


### Importando Json

In [76]:
data_churn = pd.read_json('/Telco-Customer-Churn.json')
data_churn

customerID Churn                                           customer  \
0     0002-ORFBO    No  {'gender': 'Female', 'SeniorCitizen': 0, 'Part...   
1     0003-MKNFE    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
2     0004-TLHLJ   Yes  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
3     0011-IGKFF   Yes  {'gender': 'Male', 'SeniorCitizen': 1, 'Partne...   
4     0013-EXCHZ   Yes  {'gender': 'Female', 'SeniorCitizen': 1, 'Part...   
...          ...   ...                                                ...   
7262  9987-LUTYD    No  {'gender': 'Female', 'SeniorCitizen': 0, 'Part...   
7263  9992-RRAMN   Yes  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7264  9992-UJOEL    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7265  9993-LHIEB    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7266  9995-HOTOH    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   

                                                  phone  \
0        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
1       {'PhoneService': 'Yes', 'MultipleLines': 'Yes'}   
2        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
3        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
4        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
...                                                 ...   
7262     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7263    {'PhoneService': 'Yes', 'MultipleLines': 'Yes'}   
7264     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7265     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7266  {'PhoneService': 'No', 'MultipleLines': 'No ph...   

                                               internet  \
0     {'InternetService': 'DSL', 'OnlineSecurity': '...   
1     {'InternetService': 'DSL', 'OnlineSecurity': '...   
2     {'InternetService': 'Fiber optic', 'OnlineSecu...   
3     {'InternetService': 'Fiber optic', 'OnlineSecu...   
4     {'InternetService': 'Fiber optic', 'OnlineSecu...   
...                                                 ...   
7262  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7263  {'InternetService': 'Fiber optic', 'OnlineSecu...   
7264  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7265  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7266  {'InternetService': 'DSL', 'OnlineSecurity': '...   

                                                account  
0     {'Contract': 'One year', 'PaperlessBilling': '...  
1     {'Contract': 'Month-to-month', 'PaperlessBilli...  
2     {'Contract': 'Month-to-month', 'PaperlessBilli...  
3     {'Contract': 'Month-to-month', 'PaperlessBilli...  
4     {'Contract': 'Month-to-month', 'PaperlessBilli...  
...                                                 ...  
7262  {'Contract': 'One year', 'PaperlessBilling': '...  
7263  {'Contract': 'Month-to-month', 'PaperlessBilli...  
7264  {'Contract': 'Month-to-month', 'PaperlessBilli...  
7265  {'Contract': 'Two year', 'PaperlessBilling': '...  
7266  {'Contract': 'Two year', 'PaperlessBilling': '...  

[7267 rows x 6 columns]

As váriaveis customer, phone, internet e account estão em formato JSON, sendo necessário a transformação deles para tabelas.

### Transformando dados em Json para uma tabela

In [77]:
customer_data = pd.json_normalize(data = data_churn.customer, sep='_',max_level = 1)
customer_data

gender  SeniorCitizen Partner Dependents  tenure
0     Female              0     Yes        Yes       9
1       Male              0      No         No       9
2       Male              0      No         No       4
3       Male              1     Yes         No      13
4     Female              1     Yes         No       3
...      ...            ...     ...        ...     ...
7262  Female              0      No         No      13
7263    Male              0     Yes         No      22
7264    Male              0      No         No       2
7265    Male              0     Yes        Yes      67
7266    Male              0     Yes        Yes      63

[7267 rows x 5 columns]

In [78]:
phone_data = pd.json_normalize(data = data_churn.phone, sep='_',max_level = 1)
phone_data

PhoneService     MultipleLines
0             Yes                No
1             Yes               Yes
2             Yes                No
3             Yes                No
4             Yes                No
...           ...               ...
7262          Yes                No
7263          Yes               Yes
7264          Yes                No
7265          Yes                No
7266           No  No phone service

[7267 rows x 2 columns]

In [79]:
internet_data = pd.json_normalize(data = data_churn.internet, sep='_',max_level = 1)
internet_data

InternetService OnlineSecurity OnlineBackup DeviceProtection TechSupport  \
0                DSL             No          Yes               No         Yes   
1                DSL             No           No               No          No   
2        Fiber optic             No           No              Yes          No   
3        Fiber optic             No          Yes              Yes          No   
4        Fiber optic             No           No               No         Yes   
...              ...            ...          ...              ...         ...   
7262             DSL            Yes           No               No         Yes   
7263     Fiber optic             No           No               No          No   
7264             DSL             No          Yes               No          No   
7265             DSL            Yes           No              Yes         Yes   
7266             DSL            Yes          Yes              Yes          No   

     StreamingTV StreamingMovies  
0            Yes              No  
1             No             Yes  
2             No              No  
3            Yes             Yes  
4            Yes              No  
...          ...             ...  
7262          No              No  
7263          No             Yes  
7264          No              No  
7265          No             Yes  
7266         Yes             Yes  

[7267 rows x 7 columns]

In [80]:
account_data = pd.json_normalize(data = data_churn.account, sep='_', max_level = 1)
account_data

Contract PaperlessBilling     PaymentMethod  Charges_Monthly  \
0           One year              Yes      Mailed check            65.60   
1     Month-to-month               No      Mailed check            59.90   
2     Month-to-month              Yes  Electronic check            73.90   
3     Month-to-month              Yes  Electronic check            98.00   
4     Month-to-month              Yes      Mailed check            83.90   
...              ...              ...               ...              ...   
7262        One year               No      Mailed check            55.15   
7263  Month-to-month              Yes  Electronic check            85.10   
7264  Month-to-month              Yes      Mailed check            50.30   
7265        Two year               No      Mailed check            67.85   
7266        Two year               No  Electronic check            59.00   

     Charges_Total  
0            593.3  
1            542.4  
2           280.85  
3          1237.85  
4            267.4  
...            ...  
7262         742.9  
7263        1873.7  
7264         92.75  
7265       4627.65  
7266        3707.6  

[7267 rows x 5 columns]

In [81]:
data_churn = pd.concat([data_churn,customer_data,phone_data,internet_data,account_data], axis = 1)
data_churn

customerID Churn                                           customer  \
0     0002-ORFBO    No  {'gender': 'Female', 'SeniorCitizen': 0, 'Part...   
1     0003-MKNFE    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
2     0004-TLHLJ   Yes  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
3     0011-IGKFF   Yes  {'gender': 'Male', 'SeniorCitizen': 1, 'Partne...   
4     0013-EXCHZ   Yes  {'gender': 'Female', 'SeniorCitizen': 1, 'Part...   
...          ...   ...                                                ...   
7262  9987-LUTYD    No  {'gender': 'Female', 'SeniorCitizen': 0, 'Part...   
7263  9992-RRAMN   Yes  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7264  9992-UJOEL    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7265  9993-LHIEB    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   
7266  9995-HOTOH    No  {'gender': 'Male', 'SeniorCitizen': 0, 'Partne...   

                                                  phone  \
0        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
1       {'PhoneService': 'Yes', 'MultipleLines': 'Yes'}   
2        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
3        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
4        {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
...                                                 ...   
7262     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7263    {'PhoneService': 'Yes', 'MultipleLines': 'Yes'}   
7264     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7265     {'PhoneService': 'Yes', 'MultipleLines': 'No'}   
7266  {'PhoneService': 'No', 'MultipleLines': 'No ph...   

                                               internet  \
0     {'InternetService': 'DSL', 'OnlineSecurity': '...   
1     {'InternetService': 'DSL', 'OnlineSecurity': '...   
2     {'InternetService': 'Fiber optic', 'OnlineSecu...   
3     {'InternetService': 'Fiber optic', 'OnlineSecu...   
4     {'InternetService': 'Fiber optic', 'OnlineSecu...   
...                                                 ...   
7262  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7263  {'InternetService': 'Fiber optic', 'OnlineSecu...   
7264  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7265  {'InternetService': 'DSL', 'OnlineSecurity': '...   
7266  {'InternetService': 'DSL', 'OnlineSecurity': '...   

                                                account  gender  \
0     {'Contract': 'One year', 'PaperlessBilling': '...  Female   
1     {'Contract': 'Month-to-month', 'PaperlessBilli...    Male   
2     {'Contract': 'Month-to-month', 'PaperlessBilli...    Male   
3     {'Contract': 'Month-to-month', 'PaperlessBilli...    Male   
4     {'Contract': 'Month-to-month', 'PaperlessBilli...  Female   
...                                                 ...     ...   
7262  {'Contract': 'One year', 'PaperlessBilling': '...  Female   
7263  {'Contract': 'Month-to-month', 'PaperlessBilli...    Male   
7264  {'Contract': 'Month-to-month', 'PaperlessBilli...    Male   
7265  {'Contract': 'Two year', 'PaperlessBilling': '...    Male   
7266  {'Contract': 'Two year', 'PaperlessBilling': '...    Male   

      SeniorCitizen Partner Dependents  tenure PhoneService     MultipleLines  \
0                 0     Yes        Yes       9          Yes                No   
1                 0      No         No       9          Yes               Yes   
2                 0      No         No       4          Yes                No   
3                 1     Yes         No      13          Yes                No   
4                 1     Yes         No       3          Yes                No   
...             ...     ...        ...     ...          ...               ...   
7262              0      No         No      13          Yes                No   
7263              0     Yes         No      22          Yes               Yes   
7264              0      No         No       2          Yes                No   
7265              0     Yes        Yes      67  

In [82]:
data_churn.drop(['customer','phone','internet','account'], axis=1, inplace=True)
data_churn

customerID Churn  gender  SeniorCitizen Partner Dependents  tenure  \
0     0002-ORFBO    No  Female              0     Yes        Yes       9   
1     0003-MKNFE    No    Male              0      No         No       9   
2     0004-TLHLJ   Yes    Male              0      No         No       4   
3     0011-IGKFF   Yes    Male              1     Yes         No      13   
4     0013-EXCHZ   Yes  Female              1     Yes         No       3   
...          ...   ...     ...            ...     ...        ...     ...   
7262  9987-LUTYD    No  Female              0      No         No      13   
7263  9992-RRAMN   Yes    Male              0     Yes         No      22   
7264  9992-UJOEL    No    Male              0      No         No       2   
7265  9993-LHIEB    No    Male              0     Yes        Yes      67   
7266  9995-HOTOH    No    Male              0     Yes        Yes      63   

     PhoneService     MultipleLines InternetService OnlineSecurity  \
0             Yes                No             DSL             No   
1             Yes               Yes             DSL             No   
2             Yes                No     Fiber optic             No   
3             Yes                No     Fiber optic             No   
4             Yes                No     Fiber optic             No   
...           ...               ...             ...            ...   
7262          Yes                No             DSL            Yes   
7263          Yes               Yes     Fiber optic             No   
7264          Yes                No             DSL             No   
7265          Yes                No             DSL            Yes   
7266           No  No phone service             DSL            Yes   

     OnlineBackup DeviceProtection TechSupport StreamingTV StreamingMovies  \
0             Yes               No         Yes         Yes              No   
1              No               No          No          No             Yes   
2              No              Yes          No          No              No   
3             Yes              Yes          No         Yes             Yes   
4              No               No         Yes         Yes              No   
...           ...              ...         ...         ...             ...   
7262           No               No         Yes          No              No   
7263           No               No          No          No             Yes   
7264          Yes               No          No          No              No   
7265           No              Yes         Yes          No             Yes   
7266          Yes              Yes          No         Yes             Yes   

            Contract PaperlessBilling     PaymentMethod  Charges_Monthly  \
0           One year              Yes      Mailed check            65.60   
1     Month-to-month               No      Mailed check            59.90   
2     Month-to-month              Yes  Electronic check            73.90   
3     Month-to-month              Yes  Electronic check            98.00   
4     Month-to-month              Yes      Mailed check            83.90   
...              ...              ...               ...              ...   
7262        One year               No      Mailed check            55.15   
7263  Month-to-month              Yes  Electronic check            85.10   
7264  Month-to-month              Yes      Mailed check            50.30   
7265        Two year               No      Mailed check            67.85   
7266        Two year               No  Electronic check            59.00   

     Charges_Total  
0            593.3  
1            542.4  
2           280.85  
3          1237.85  
4            267.4  
...            ...  
7262         742.9  
7263        1873.7  
7264         92.75  
7265       4627.65  
7266        3707.6  

[7267 rows x 21 columns]

Transformado o JSON para tabelas utilizando json_normalize e concatenando com a tabela principal, depois excluindo as colunas onde estavam esses arquivos JSON.

### Analisando dados

#### Verificando colunas e informações gerais do dataframe

In [83]:
data_churn.columns

Index(['customerID', 'Churn', 'gender', 'SeniorCitizen', 'Partner',
       'Dependents', 'tenure', 'PhoneService', 'MultipleLines',
       'InternetService', 'OnlineSecurity', 'OnlineBackup', 'DeviceProtection',
       'TechSupport', 'StreamingTV', 'StreamingMovies', 'Contract',
       'PaperlessBilling', 'PaymentMethod', 'Charges_Monthly',
       'Charges_Total'],
      dtype='object')

In [84]:
data_churn.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7267 entries, 0 to 7266
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7267 non-null   object 
 1   Churn             7267 non-null   object 
 2   gender            7267 non-null   object 
 3   SeniorCitizen     7267 non-null   int64  
 4   Partner           7267 non-null   object 
 5   Dependents        7267 non-null   object 
 6   tenure            7267 non-null   int64  
 7   PhoneService      7267 non-null   object 
 8   MultipleLines     7267 non-null   object 
 9   InternetService   7267 non-null   object 
 10  OnlineSecurity    7267 non-null   object 
 11  OnlineBackup      7267 non-null   object 
 12  DeviceProtection  7267 non-null   object 
 13  TechSupport       7267 non-null   object 
 14  StreamingTV       7267 non-null   object 
 15  StreamingMovies   7267 non-null   object 
 16  Contract          7267 non-null   object 


Verificando as colunas pelo info(), é possível perceber que não há nenhuma váriavel nula, além disso, há alguns problemas em relação a padronização dos nomes das colunas, com nome iniciando em letra minúscula (e.g, gender) e palavras separadas com _ como Charges_Monthly, além de uma possível tipagem errada em Charges_Total

#### Analisando individualmente as colunas

##### Coluna CustomerID

In [85]:
data_churn.customerID.unique()

array(['0002-ORFBO', '0003-MKNFE', '0004-TLHLJ', ..., '9992-UJOEL',
       '9993-LHIEB', '9995-HOTOH'], dtype=object)

In [86]:
data_churn.rename(columns = {'customerID':'CustomerID'},inplace=True)

In [87]:
data_churn.CustomerID.unique()

array(['0002-ORFBO', '0003-MKNFE', '0004-TLHLJ', ..., '9992-UJOEL',
       '9993-LHIEB', '9995-HOTOH'], dtype=object)

##### Coluna Churn

In [88]:
data_churn.Churn.unique()

array(['No', 'Yes', ''], dtype=object)

In [89]:
data_churn.Churn.replace('','No info', inplace = True)

In [90]:
data_churn.Churn.unique()

array(['No', 'Yes', 'No info'], dtype=object)

##### Coluna Gender

In [91]:
data_churn.gender.unique()

array(['Female', 'Male'], dtype=object)

In [92]:
data_churn.rename(columns = {'gender':'Gender'},inplace=True)

In [93]:
data_churn.Gender.unique()

array(['Female', 'Male'], dtype=object)

##### Coluna SeniorCitizen

In [94]:
data_churn.SeniorCitizen.unique()

array([0, 1])

##### Coluna Partner

In [95]:
data_churn.Partner.unique()

array(['Yes', 'No'], dtype=object)

##### Coluna Dependents

In [96]:
data_churn.Dependents.unique()

array(['Yes', 'No'], dtype=object)

##### Coluna Tenure

In [97]:
data_churn.tenure.unique()

array([ 9,  4, 13,  3, 71, 63,  7, 65, 54, 72,  5, 56, 34,  1, 45, 50, 23,
       55, 26, 69, 11, 37, 49, 66, 67, 20, 43, 59, 12, 27,  2, 25, 29, 14,
       35, 64, 39, 40,  6, 30, 70, 57, 58, 16, 32, 33, 10, 21, 61, 15, 44,
       22, 24, 19, 47, 62, 46, 52,  8, 60, 48, 28, 41, 53, 68, 51, 31, 36,
       17, 18, 38, 42,  0])

In [98]:
data_churn.rename(columns = {'tenure':'Tenure'},inplace=True)

In [99]:
data_churn.Tenure.unique()

array([ 9,  4, 13,  3, 71, 63,  7, 65, 54, 72,  5, 56, 34,  1, 45, 50, 23,
       55, 26, 69, 11, 37, 49, 66, 67, 20, 43, 59, 12, 27,  2, 25, 29, 14,
       35, 64, 39, 40,  6, 30, 70, 57, 58, 16, 32, 33, 10, 21, 61, 15, 44,
       22, 24, 19, 47, 62, 46, 52,  8, 60, 48, 28, 41, 53, 68, 51, 31, 36,
       17, 18, 38, 42,  0])

##### Coluna PhoneService

In [100]:
data_churn.PhoneService.unique()

array(['Yes', 'No'], dtype=object)

##### Coluna MultipleLines

In [101]:
data_churn.MultipleLines.unique()

array(['No', 'Yes', 'No phone service'], dtype=object)

##### Coluna InternetService

In [102]:
data_churn.InternetService.unique()

array(['DSL', 'Fiber optic', 'No'], dtype=object)

##### Coluna OnlineSecurity

In [103]:
data_churn.OnlineSecurity.unique()

array(['No', 'Yes', 'No internet service'], dtype=object)

##### Coluna OnlineBackup

In [104]:
data_churn.OnlineBackup.unique()

array(['Yes', 'No', 'No internet service'], dtype=object)

##### Coluna DeviceProtection

In [105]:
data_churn.DeviceProtection.unique()

array(['No', 'Yes', 'No internet service'], dtype=object)

##### Coluna TechSupport

In [106]:
data_churn.TechSupport.unique()

array(['Yes', 'No', 'No internet service'], dtype=object)

##### Coluna StreamingTV

In [107]:
data_churn.StreamingTV.unique()

array(['Yes', 'No', 'No internet service'], dtype=object)

##### Coluna StreamingMovies

In [108]:
data_churn.StreamingMovies.unique()

array(['No', 'Yes', 'No internet service'], dtype=object)

##### Coluna Contract

In [109]:
data_churn.Contract.unique()

array(['One year', 'Month-to-month', 'Two year'], dtype=object)

##### Coluna PaperlessBilling

In [110]:
data_churn.PaperlessBilling.unique()

array(['Yes', 'No'], dtype=object)

##### Coluna PaymentMethod

In [111]:
data_churn.PaymentMethod.unique()

array(['Mailed check', 'Electronic check', 'Credit card (automatic)',
       'Bank transfer (automatic)'], dtype=object)

##### Coluna ChargesMonthly

In [112]:
data_churn.Charges_Monthly.unique()  

array([65.6 , 59.9 , 73.9 , ..., 91.75, 68.8 , 67.85])

In [113]:
data_churn.rename(columns = {'Charges_Monthly':'ChargesMonthly'},inplace=True)

In [114]:
data_churn.ChargesMonthly.unique()

array([65.6 , 59.9 , 73.9 , ..., 91.75, 68.8 , 67.85])

##### Coluna ChargesTotal

In [115]:
data_churn.Charges_Total.unique()

array(['593.3', '542.4', '280.85', ..., '742.9', '4627.65', '3707.6'],
      dtype=object)

In [116]:
data_churn.rename(columns = {'Charges_Total':'ChargesTotal'},inplace=True)

In [117]:
data_churn.ChargesTotal.unique()

array(['593.3', '542.4', '280.85', ..., '742.9', '4627.65', '3707.6'],
      dtype=object)

In [118]:
data_churn.ChargesTotal = pd.to_numeric(data_churn.ChargesTotal, downcast="float")

ValueError: ignored

In [119]:
data_churn.ChargesTotal.replace(' ','0', inplace = True)

In [120]:
data_churn.ChargesTotal = data_churn.ChargesTotal.astype(np.float64)

In [121]:
data_churn.ChargesTotal.unique()

array([ 593.3 ,  542.4 ,  280.85, ...,  742.9 , 4627.65, 3707.6 ])

Nessa etapa, foi necessário alterar o nome das colunas que não estavam no padrão, além de tratar dados que estavam adicionados de forma errada, como na coluna Churn, onde foi feito um replace dos dados que estavam vazios para o nome 'No Info'. Foi notado também que na tabela ChargesTotal, existia dados com o valor ' ', foi feito o replace para o valor zero, pois essa informação mostrava que ainda não tinha sido pago nenhum tipo de valor à empresa.

Em relação ao tipo de dados, foi feito a divisão da seguinte maneira:
  * Quantitativa:
    * Discreta:
      1. Tenure
    * Contínua:
      1. ChargesTotal
      2. ChargesMonthly
  * Qualitativa
    * Nominal:
      1. CustomerID
      2. Churn
      3. Gender
      4. SeniorCitizen
      5. Partner
      6. Dependents
      7. PhoneService
      8. InternetService
      9. OnlineSecurity
      10. OnlineBackup
      11. DeviceProtection
      12. TechSupport
      13. StreamingTV
      14. StreamingMovies
      15. Contract
      16. PaperlessBilling
      17. PaymentMethod

### Extras Primeira Semana


#### Coluna de gasto diário

In [122]:
chargesDaily = round(data_churn.ChargesMonthly / 30,2)
chargesDaily

0       2.19
1       2.00
2       2.46
3       3.27
4       2.80
        ... 
7262    1.84
7263    2.84
7264    1.68
7265    2.26
7266    1.97
Name: ChargesMonthly, Length: 7267, dtype: float64

In [123]:
data_churn.insert(18,"ChargesDaily",chargesDaily)

In [124]:
data_churn

CustomerID Churn  Gender  SeniorCitizen Partner Dependents  Tenure  \
0     0002-ORFBO    No  Female              0     Yes        Yes       9   
1     0003-MKNFE    No    Male              0      No         No       9   
2     0004-TLHLJ   Yes    Male              0      No         No       4   
3     0011-IGKFF   Yes    Male              1     Yes         No      13   
4     0013-EXCHZ   Yes  Female              1     Yes         No       3   
...          ...   ...     ...            ...     ...        ...     ...   
7262  9987-LUTYD    No  Female              0      No         No      13   
7263  9992-RRAMN   Yes    Male              0     Yes         No      22   
7264  9992-UJOEL    No    Male              0      No         No       2   
7265  9993-LHIEB    No    Male              0     Yes        Yes      67   
7266  9995-HOTOH    No    Male              0     Yes        Yes      63   

     PhoneService     MultipleLines InternetService OnlineSecurity  \
0             Yes                No             DSL             No   
1             Yes               Yes             DSL             No   
2             Yes                No     Fiber optic             No   
3             Yes                No     Fiber optic             No   
4             Yes                No     Fiber optic             No   
...           ...               ...             ...            ...   
7262          Yes                No             DSL            Yes   
7263          Yes               Yes     Fiber optic             No   
7264          Yes                No             DSL             No   
7265          Yes                No             DSL            Yes   
7266           No  No phone service             DSL            Yes   

     OnlineBackup DeviceProtection TechSupport StreamingTV StreamingMovies  \
0             Yes               No         Yes         Yes              No   
1              No               No          No          No             Yes   
2              No              Yes          No          No              No   
3             Yes              Yes          No         Yes             Yes   
4              No               No         Yes         Yes              No   
...           ...              ...         ...         ...             ...   
7262           No               No         Yes          No              No   
7263           No               No          No          No             Yes   
7264          Yes               No          No          No              No   
7265           No              Yes         Yes          No             Yes   
7266          Yes              Yes          No         Yes             Yes   

            Contract PaperlessBilling  ChargesDaily     PaymentMethod  \
0           One year              Yes          2.19      Mailed check   
1     Month-to-month               No          2.00      Mailed check   
2     Month-to-month              Yes          2.46  Electronic check   
3     Month-to-month              Yes          3.27  Electronic check   
4     Month-to-month              Yes          2.80      Mailed check   
...              ...              ...           ...               ...   
7262        One year               No          1.84      Mailed check   
7263  Month-to-month              Yes          2.84  Electronic check   
7264  Month-to-month              Yes          1.68      Mailed check   
7265        Two year               No          2.26      Mailed check   
7266        Two year               No          1.97  Electronic check   

      ChargesMonthly  ChargesTotal  
0              65.60        593.30  
1              59.90        542.40  
2              73.90        280.85  
3              98.00       1237.85  
4              83.90        267.40  
...              ...           ...  
7262           55.15        742.90  
7263           85.10       1873.70  
7264           50.30         92.75  
7265           67.85       4627.65  
7266           59.00       3707

#### Tradução das Colunas

In [125]:
colunas_traduzidas = {
    'CustomerID':'ClienteID',
    'Churn':'SaidaCliente',
    'Gender':'Sexo',
    'SeniorCitizen':'Idoso',
    'Partner': 'Parceirx',
    'Dependents': 'Dependentes',
    'Tenure': 'MesesContrato',
    'PhoneService': 'AssinaturaTelefone',
    'MultipleLines': 'MuitasAssinaturasTelefone',
    'InternetService': 'ServicoInternet',
    'OnlineSecurity': 'SegurancaOnline',
    'OnlineBackup': 'BackupAdicional',
    'DeviceProtection': 'ProtecaoDispositivo',
    'TechSupport': 'SuporteTecnico',
    'StreamingTV': 'TVACabo',
    'StreamingMovies': 'ServicoStreaming',
    'Contract': 'Contrato',
    'PaperlessBilling': 'PagamentoOnline',
    'PaymentMethod': 'FormaPagamento',
    'ChargesMonthly': 'CobrancaMensal',
    'ChargesTotal': 'CobrancaTotal',
    'ChargesDaily': 'CobrancaDiaria'
}

In [126]:
data_churn_traduzido = data_churn.rename(columns=colunas_traduzidas)

In [127]:
data_churn_traduzido

ClienteID SaidaCliente    Sexo  Idoso Parceirx Dependentes  \
0     0002-ORFBO           No  Female      0      Yes         Yes   
1     0003-MKNFE           No    Male      0       No          No   
2     0004-TLHLJ          Yes    Male      0       No          No   
3     0011-IGKFF          Yes    Male      1      Yes          No   
4     0013-EXCHZ          Yes  Female      1      Yes          No   
...          ...          ...     ...    ...      ...         ...   
7262  9987-LUTYD           No  Female      0       No          No   
7263  9992-RRAMN          Yes    Male      0      Yes          No   
7264  9992-UJOEL           No    Male      0       No          No   
7265  9993-LHIEB           No    Male      0      Yes         Yes   
7266  9995-HOTOH           No    Male      0      Yes         Yes   

      MesesContrato AssinaturaTelefone MuitasAssinaturasTelefone  \
0                 9                Yes                        No   
1                 9                Yes                       Yes   
2                 4                Yes                        No   
3                13                Yes                        No   
4                 3                Yes                        No   
...             ...                ...                       ...   
7262             13                Yes                        No   
7263             22                Yes                       Yes   
7264              2                Yes                        No   
7265             67                Yes                        No   
7266             63                 No          No phone service   

     ServicoInternet SegurancaOnline BackupAdicional ProtecaoDispositivo  \
0                DSL              No             Yes                  No   
1                DSL              No              No                  No   
2        Fiber optic              No              No                 Yes   
3        Fiber optic              No             Yes                 Yes   
4        Fiber optic              No              No                  No   
...              ...             ...             ...                 ...   
7262             DSL             Yes              No                  No   
7263     Fiber optic              No              No                  No   
7264             DSL              No             Yes                  No   
7265             DSL             Yes              No                 Yes   
7266             DSL             Yes             Yes                 Yes   

     SuporteTecnico TVACabo ServicoStreaming        Contrato PagamentoOnline  \
0               Yes     Yes               No        One year             Yes   
1                No      No              Yes  Month-to-month              No   
2                No      No               No  Month-to-month             Yes   
3                No     Yes              Yes  Month-to-month             Yes   
4               Yes     Yes               No  Month-to-month             Yes   
...             ...     ...              ...             ...             ...   
7262            Yes      No               No        One year              No   
7263             No      No              Yes  Month-to-month             Yes   
7264             No      No               No  Month-to-month             Yes   
7265            Yes      No              Yes        Two year              No   
7266             No     Yes              Yes        Two year              No   

      CobrancaDiaria    FormaPagamento  CobrancaMensal  CobrancaTotal  
0               2.19      Mailed check           65.60         593.30  
1               2.00      Mailed check           59.90         542.40  
2               2.46  Electronic check           73.90         280.85  
3               3.27  Electronic check           98.00        1237.85  
4               2.80      Mailed check           83.90         267.40  
...              ...               ...             ... 

### Estado da tabela da primeira semana

In [128]:
data_churn

CustomerID Churn  Gender  SeniorCitizen Partner Dependents  Tenure  \
0     0002-ORFBO    No  Female              0     Yes        Yes       9   
1     0003-MKNFE    No    Male              0      No         No       9   
2     0004-TLHLJ   Yes    Male              0      No         No       4   
3     0011-IGKFF   Yes    Male              1     Yes         No      13   
4     0013-EXCHZ   Yes  Female              1     Yes         No       3   
...          ...   ...     ...            ...     ...        ...     ...   
7262  9987-LUTYD    No  Female              0      No         No      13   
7263  9992-RRAMN   Yes    Male              0     Yes         No      22   
7264  9992-UJOEL    No    Male              0      No         No       2   
7265  9993-LHIEB    No    Male              0     Yes        Yes      67   
7266  9995-HOTOH    No    Male              0     Yes        Yes      63   

     PhoneService     MultipleLines InternetService OnlineSecurity  \
0             Yes                No             DSL             No   
1             Yes               Yes             DSL             No   
2             Yes                No     Fiber optic             No   
3             Yes                No     Fiber optic             No   
4             Yes                No     Fiber optic             No   
...           ...               ...             ...            ...   
7262          Yes                No             DSL            Yes   
7263          Yes               Yes     Fiber optic             No   
7264          Yes                No             DSL             No   
7265          Yes                No             DSL            Yes   
7266           No  No phone service             DSL            Yes   

     OnlineBackup DeviceProtection TechSupport StreamingTV StreamingMovies  \
0             Yes               No         Yes         Yes              No   
1              No               No          No          No             Yes   
2              No              Yes          No          No              No   
3             Yes              Yes          No         Yes             Yes   
4              No               No         Yes         Yes              No   
...           ...              ...         ...         ...             ...   
7262           No               No         Yes          No              No   
7263           No               No          No          No             Yes   
7264          Yes               No          No          No              No   
7265           No              Yes         Yes          No             Yes   
7266          Yes              Yes          No         Yes             Yes   

            Contract PaperlessBilling  ChargesDaily     PaymentMethod  \
0           One year              Yes          2.19      Mailed check   
1     Month-to-month               No          2.00      Mailed check   
2     Month-to-month              Yes          2.46  Electronic check   
3     Month-to-month              Yes          3.27  Electronic check   
4     Month-to-month              Yes          2.80      Mailed check   
...              ...              ...           ...               ...   
7262        One year               No          1.84      Mailed check   
7263  Month-to-month              Yes          2.84  Electronic check   
7264  Month-to-month              Yes          1.68      Mailed check   
7265        Two year               No          2.26      Mailed check   
7266        Two year               No          1.97  Electronic check   

      ChargesMonthly  ChargesTotal  
0              65.60        593.30  
1              59.90        542.40  
2              73.90        280.85  
3              98.00       1237.85  
4              83.90        267.40  
...              ...           ...  
7262           55.15        742.90  
7263           85.10       1873.70  
7264           50.30         92.75  
7265           67.85       4627.65  
7266           59.00       3707

### Resumo
  Em resumo:
  * Feita a tradução das colunas do nosso dataset;
  * Dados com problemas foram tratados;
  * Coluna ChargesTotal convertida de string pra float64;
  * Campo ChargesDaily adicionados

  No final, o Dataset possui 22 colunas com 7267 linhas.